_posts/2018-03-19-making-a-components-pod.markdown

```diff
@@ -0,0 +1,486 @@
+
+$ ls src/
+App.js    __tests__ app.json  index.js
+```
+
+To ensure everything is still hooked up, let's make sure that all of your tests are working in the new repo.
+
+```sh
+$ yarn test
+
+yarn run v1.5.1
+$ jest
+ PASS  src/__tests__/App.js
+  ✓ renders correctly (176ms)
+
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+Snapshots:   0 total
+Time:        1.392s
+Ran all test suites.
+✨  Done in 2.32s.
+```
+
+We're now going to be done with our JavaScript side, basically is our React Native "hello world". It's a React Native
+project that exposes a single component which says `"Welcome to React Native!"`.
+
+### Deployment
+
+We're going to want to have this exposed to our native libraries, so we're going to ship the bundled JavaScript as our
+library's source code. We do this via the React Native CLI, and it's going to place the file inside our Pod folder from
+earlier.
+
+```sh
+$ react-native bundle --entry-file src/index.js --bundle-output Pod/Assets/GitDawg.js --assets-dest Pod/Assets
+```
+
+## GitDawg Pod
+
+With that done, we can start looking at the native side of our codebase. We let `pod lib create` set up an Example app
+for us to work with in the repo, which consumes a Podspec in the root. So we're going to take a look at the Podspec, and
+update it.
+
+Our goal with the Example app is to set up an app exclusively for developing components in. In Artsy's case, this app
+handles auth and has a series of jump-off points for developing a component (either through Storybooks or directly using
+the Pods' `UIViewController`s.)
+
+To get started we need to modify the CocoaPod this repo represents:
+
+* Update our Podspec to handle React Native as a dependency, and our assets
+* Add support for native compilation via CocoaPods with [cocoapods-fix-react-native][cpfrn]
+* Create a single `UIViewController` subclass for the Welcome Screen using the bundled React Native JS
+
+We want to have our Podspec re-use the metadata from React Native to set up GitDawg's dependencies. So replace
+`GitDawg.podspec` with this:
+
+```ruby
+require 'json'
+
+# Returns the version number for a package.json file
+pkg_version = lambda do |dir_from_root = '', version = 'version'|
+  path = File.join(__dir__, dir_from_root, 'package.json')
+  JSON.parse(File.read(path))[version]
+end
+
+# Let the main package.json decide the version number for the pod
+gitdawg_version = pkg_version.call
+# Use the same RN version that the JS tools use
+react_native_version = pkg_version.call('node_modules/react-native')
+
+Pod::Spec.new do |s|
+  s.name             = 'GitDawg'
+  s.version          = gitdawg_version
+  s.description      = 'Components for GitHawk.'
+  s.homepage         = 'https://github.com/orta/GitDawg'
+  s.license          = { type: 'MIT', file: 'LICENSE' }
+  s.author           = { 'orta' => 'orta.therox@gmail.com' }
+  s.source           = { git: 'https://github.com/orta/GitDawg.git', tag: s.version.to_s }
+
```

Comment:
Critically, we need to include `s.platform = :ios, '9.0'` or similar.

Reply:
I think we can probably detect this in the fixer also, so I'm gonna do it

Reply:
Never make the same mistake twice

<img width="1151" alt="screen shot 2018-04-07 at 2 09 56 pm" src="https://user-images.githubusercontent.com/49038/38458548-5fd6d080-3a6d-11e8-8829-d04b2e396076.png">